jme3-core/src/main/java/com/jme3/anim/MorphControl.java

```diff
@@ -79,22 +82,31 @@
     private static final VertexBuffer.Type bufferTypes[] = VertexBuffer.Type.values();
 
     @Override
-    protected void controlUpdate(float tpf) {
-        if (!enabled) {
-            return;
+    public void setSpatial(Spatial spatial) {
+        super.setSpatial(spatial);
+
+        // Remove matparam override from the old targets
+        for (Geometry target : targets.getArray()) {
+            target.removeMatParamOverride(nullNumberOfBones);
         }
+
         // gathering geometries in the sub graph.
-        // This must be done in the update phase as the gathering might add a matparam override
+        // This must be done in the update phase as the gathering might add a matparam override otherwise it will
```

Comment:
I don't understand the comment about "must be done in the update phase". Can you clarify?

Reply:
My understanding is that the author means this should not be done in `controlRender()` because it can add a MatParamOverride which can throw IllegalStateException if added in the render phase. 

See: https://hub.jmonkeyengine.org/t/morphcontrol-question/44599/8?u=ali_rs

Reply:
That made sense when it was in `controlUpdate()`. But it's in `setSpatial()` now, so it's not being done in the update phase.

Reply:
I have tested this on bunch of different models and I did not noticed any issue when done in `setSpatial()`.

so maybe I should remove that comment or change it to something like this:

```
// This must not be done in the render phase as the gathering might add a matparam  
// override which will cause an IllegalStateException if done in the render phase.
```

what do you think?

Reply:
That should be fine.

Reply:
Do you mean using the flag?

Reply:
I meant changing the comment. I don't see the need for a flag.

Reply:
I see. Will update the comment.

Reply:
Fixed the comment.